### DISCUSSION TITLE
zgenom update zcompile error

### DISCUSSION THREAD

**Discussion by mlhynfield** [jandamm/zgenom#121] [Q&A] [answered]
I recently noticed that whenever I run `zgenom update` and open a new terminal, I get the following error:

`__zgenom_compile:zcompile:2: can't write zwc file: /Users/mlhynfield/.zcompdump-MHYNFIELD – LAPTOP-5.8.zwc`

I'm guessing this comes from the `zgenom-compile -U $ZGEN_CUSTOM_COMPDUMP` command in zgenom-save (based on some testing).

It seems like the referenced .zwc file exists, so it may be created elsewhere anyway, but is there anything I can do to remove the error?

**Comment by jandamm** [comment, marked as answer]
Thanks for reporting.
The issue is that you have an invalid character in your computer name: `–`.
While it looks like a `-` it is a bit longer and cannot be passed to `zcompile` as output name.

There have been similar reports with the default name of macs where a `’` is contained. See https://github.com/jandamm/zgenom/issues/100

I haven't gotten around to it but I plan to move the zcompdump generated by zgenom to `zgenom/sources` so this issue won't happen again.

For now you could use `$ZGEN_CUSTOM_COMPDUMP` to overwrite the default name. Ideally you'd remove this variable when I add the fix.

**Reply by mlhynfield** [reply]
Thank you for the response, will do!

**Reply by jandamm** [reply]
@mlhynfield if you update to the newest version of `zgenom` you should be able to remove `$ZGEN_CUSTOM_COMPDUMP` and the issue should be gone. 👍
